Sources/MessagingInApp/MessagingInAppImplementation.swift

```diff
@@ -59,6 +59,14 @@ class MessagingInAppImplementation: MessagingInAppInstance {
         self.eventListener = eventListener
     }
 
+    // Sets a custom event bus handler for unit testing.
+    // Do not use it in production.
+    #if DEBUG
+    func setEventBusHandler(_ eventBusHandler: EventBusHandler) {
+        self.eventBusHandler = eventBusHandler
+    }
+    #endif
+
```

Comment:
For some reason I am unsure of, majority of the test cases fail on using `DIGraphShared.shared.override(value: eventBusHandlerMock, forType: EventBusHandler.self)` in `setup` method and this is why I avoided to use the above in setup.

Reply:
Now that I think about it, with [the test class moving towards an integration test](https://github.com/customerio/customerio-ios/pull/776), I don't think it's a good idea to mock the eventbus. We want the eventbus to post real events since it's posting more then push metrics. 

Another idea [is to add another observer to the real eventbus instance](https://github.com/customerio/customerio-ios/pull/774/commits/36bc1c72291226afb5dbc5b7753766613bd0fb8c#diff-d45d2abf8395422b4e74c4ceec60e0157a10d5489cfd9ac12d88c35bf9afec26R17-R22) so we don't need to mock the eventbus but still run assertions against the events that got posted to it.